app/javascript/components/shared/TopHeader.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import PropTypes from "prop-types";
+import TopHeaderDropdownMenu from "./TopHeaderDropdownMenu";
+
+class TopHeader extends React.Component {
+  render() {
+    const { username, isAdmin, profileLink, editUserLink, adminLink, logoutLink } = this.props;
+    const topHeaderDropdownMenuComponent = function() {
+      if (username) {
+        return (
+          <TopHeaderDropdownMenu
```

Comment:
You can use this component expressly in react render ( return () )
It just recommendation. No blocker for PR
Example:
```javascript
return (
 { username && <TopHeaderDropdownMenu /> }
)
```